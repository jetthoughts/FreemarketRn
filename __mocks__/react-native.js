const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('../app/api/firebase', () => {
  return {
    saveImageToStorage: jest.fn(),
  }
});

const helpers = require('../app/lib/helpers');
helpers.dateToTimeString = () => '10:20';

module.exports = rn;
