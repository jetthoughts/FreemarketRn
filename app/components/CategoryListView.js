import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ListView } from 'react-native';

import CategoryListRow from '../components/CategoryListRow';

export default class CategoryListView extends Component {
  constructor(props) {
    super(props);

    props.loadCategories();

    const ds = new ListView.DataSource({
      rowHasChanged: (r1, r2) => r1.id !== r2.id,
    });
    this.dataSource = ds.cloneWithRows([]);
  }

  shouldComponentUpdate(nextProps) {
    return nextProps.categories.length !== this.props.categories.length ||
      nextProps.categories.map(cat => cat.id) !== this.props.categories.map(cat => cat.id);
  }

  componentWillUpdate(nextProps) {
    this.dataSource = this.dataSource.cloneWithRows(nextProps.categories);
  }

  render() {
    const { dataSource } = this;
    const { onCategoryPress } = this.props;
    const renderRow = row => (
      <CategoryListRow onPress={() => onCategoryPress(row)} name={row.name} icon={row.icon} />
    );

    return <ListView {...{ dataSource, renderRow }} />;
  }
}

CategoryListView.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
  })).isRequired,
  loadCategories: PropTypes.func.isRequired,
  onCategoryPress: PropTypes.func.isRequired,
};
