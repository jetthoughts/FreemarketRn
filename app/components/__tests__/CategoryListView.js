import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import CategoryListView from '../CategoryListView';

it('renders successfully', () => {
  const tree = renderer.create(
    <CategoryListView categories={[]} loadCategories={() => 'true'} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
