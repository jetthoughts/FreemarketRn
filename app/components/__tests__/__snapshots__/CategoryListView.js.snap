// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 0,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
>
  <View />
</RCTScrollView>
`;
